common/models/templates/test/test-tokenization.js

```diff
@@ -0,0 +1,358 @@
+        right: " jumped over the lazy dog",
+        startOfBuffer: true,
+        endOfBuffer: true
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: ['The', 'quick', 'brown', 'fox'],
+        right: ['jumped', 'over', 'the', 'lazy', 'dog'],
+        caretSplitsToken: false
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, pre-whitespace caret, partial context', function() {
+      let context = {
+        left: "quick brown fox",        // No "The"
+        right: " jumped over the lazy", // No "dog"
+        startOfBuffer: false,
+        endOfBuffer: false
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: ['quick', 'brown', 'fox'],
+        right: ['jumped', 'over', 'the', 'lazy'],
+        caretSplitsToken: false
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, post-whitespace caret', function() {
+      let context = {
+        left: "The quick brown fox ",
+        right: "jumped over the lazy dog",
+        startOfBuffer: true,
+        endOfBuffer: true
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      // Technically, we're editing the start of the first token on the right
+      // when in this context.
+      let expectedResult = {
+        left: ['The', 'quick', 'brown', 'fox', ''],
+        right: ['jumped', 'over', 'the', 'lazy', 'dog'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, post-whitespace caret, partial context', function() {
+      let context = {
+        left: "quick brown fox ",
+        right: "jumped over the lazy",
+        startOfBuffer: false,
+        endOfBuffer: false
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      // Technically, we're editing the start of the first token on the right
+      // when in this context.
+      let expectedResult = {
+        left: ['quick', 'brown', 'fox', ''],
+        right: ['jumped', 'over', 'the', 'lazy'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, splitting caret', function() {
+      let context = {
+        left: "The quick brown fox jum",
+        right: "ped over the lazy dog",
+        startOfBuffer: true,
+        endOfBuffer: true
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: ['The', 'quick', 'brown', 'fox', 'jum'],
+        right: ['ped', 'over', 'the', 'lazy', 'dog'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
```

Comment:
Could you copy-paste this exact test case, but where the context does _not_ include the `startOfBuffer`?